src/schedule/edf_schedule.c

```diff
@@ -258,6 +258,11 @@ int scheduler_init_edf(void)
 
 	edf_sch = rzalloc(SOF_MEM_FLAG_KERNEL,
 			  sizeof(*edf_sch));
+	if (!edf_sch) {
+		tr_err(&edf_tr, "scheduler_init_edf(): allocation failed");
+		return -ENOMEM;
+	}
```

Comment:
there's an error case 13 lines below - should add an `rfree()` there

Reply:
fixed